### ISSUE TITLE
Hide audio slider, add microphone slider, add system tray, scroll through workspaces and AMD GPU 

### ISSUE THREAD

**Issue by linuxmobile** [scorpion-26/gBar#5]
The bar really consumes very little, **I loved it**. Even less than waybar, in my case.

- I think you could add a systray. Since I can see the telegram and discord icons.

- You could also add that the volume icon opens the slider with a hover and not that it is always open.

- Mic slider with hover effect to open the slider too.

- Switch Workspaces with scroll (hyprctl dispatch workspace e+1 & hyprctl dispatch workspace e-1)

(an another little things, i've got amd gpu. If you can add this too i will love you a lot <3)

**Comment by scorpion-26** [comment]
In regards to the amd gpu: 
Since I don't have an AMD gpu, can you confirm, that the files mentioned in https://unix.stackexchange.com/questions/576707/how-to-monitor-amd-gpu-on-linux-server exist? If so, can you post the output of them(with cat)?

> add a systray
That should be possible. Unfortunately, I'm currently quite busy, so it could be a while(~1 Month) until I can get around to implement such a major feature.

> You could also add that the volume icon opens the slider with a hover and not that it is always open.
That is intended behaviour, but I probably can add a config option for that. 

> Mic slider with hover effect to open the slider too.
Should be possible. Since I don't have a microphone ready, I probably need some feedback if it works.

> Switch Workspaces with scroll
Should also be possible

**Comment by linuxmobile** [comment]
> https://unix.stackexchange.com/questions/576707/how-to-monitor-amd-gpu-on-linux-server exist? If so, can you post the output of them(with cat)?

![image](https://user-images.githubusercontent.com/10554636/215353315-81580445-f77c-48ab-8ba7-4ed420dbeed6.png)


And again. Thx for the work. i love it

**Comment by scorpion-26** [comment]
Am I right to conclude, that you have 2GiB VRAM? (So I know in which unit the VRAM is measured. I assume, it is in bytes)

**Comment by linuxmobile** [comment]
> Am I right to conclude, that you have 2GiB VRAM? (So I know in which unit the VRAM is measured. I assume, it is in bytes)

Yeah, ive got 2gb

**Comment by scorpion-26** [comment]
I've just added a first implementation of the AMD GPU status. Since I can't test it, I would appreciate it, if you can test it: 
For that, can you do the following: 
Pull the latest commit

*Completely remove all build files and rebuild*

```
rm -rf build
meson setup build
ninja -C build
build/gBar bar 0
```

Everything **except the temperature** should work. If it doesn't (e.g. crash), please attach the log.

I also need some more information about the temperature:
```
cat /sys/class/drm/card0/device/hwmon/hwmon*/temp1_input
cat /sys/class/drm/card0/device/hwmon/hwmon*/temp1_label
```

Thanks!

**Comment by linuxmobile** [comment]
> cat /sys/class/drm/card0/device/hwmon/hwmon*/temp1_input
> cat /sys/class/drm/card0/device/hwmon/hwmon*/temp1_label

Gpu module is working now. Thx u <3

![image](https://user-images.githubusercontent.com/10554636/215554199-edef7605-4711-478f-a23d-3d594955784f.png)

**Comment by scorpion-26** [comment]
one last thing: can you ```ls /sys/class/drm/card0/device/hwmon/```?

**Comment by linuxmobile** [comment]
![image](https://user-images.githubusercontent.com/10554636/215578359-707c4950-e1b5-4ca1-b2f6-92a9e1426fc2.png)

Ask me what you need!

**Comment by scorpion-26** [comment]
AMD should work now, just like Nvidia

**Comment by linuxmobile** [comment]
> just like Nvidia

Let me rebuild

**Comment by scorpion-26** [comment]
> You could also add that the volume icon opens the slider with a hover and not that it is always open.
Just added that. To enable it, just add ```AudioRevealer: true``` to your config

**Comment by linuxmobile** [comment]
> > You could also add that the volume icon opens the slider with a hover and not that it is always open.
> 
> Just added that. To enable it, just add `AudioRevealer: true` to your config

Let me try!

**Comment by scorpion-26** [comment]
> Switch Workspaces with scroll (hyprctl dispatch workspace e+1 & hyprctl dispatch workspace e-1)

Just implemented the workspace scroll. By default it calls now m+1 and m-1, but that can be changed by setting ```WorkspaceScrollOnMonitor: false``` in the config (Then it'll call e+1 and e-1). The scroll direction can be inverted using ```WorkspaceScrollInvert: true```.

**Comment by linuxmobile** [comment]
> build/gBar bar 0

I do a rebuild
```
rm -rf build
meson setup build
ninja -C build
build/gBar bar 0
```
and now i got this issue: 
![image](https://user-images.githubusercontent.com/10554636/218332492-20e177fa-2864-4942-8201-4bb1761bbb0a.png)

**Comment by scorpion-26** [comment]
Oh, you need to either set the name of your network adapter in the config ("NetworkAdapter") or disable it by setting NetworkWidget to false in the config. 
I forgot to do a check, my bad.

**Comment by scorpion-26** [comment]
It should work now on the latest commit, even without changing the config

> ```meson setup build```
By the way, you probably want to run ```meson setup build -Dbuildtype=release``` for everyday use builds, as it compiles with optimizations.

**Comment by linuxmobile** [comment]
> NetworkWidget

**Oh. I forgot to tell you yes. Everything works perfect!**

_Just to update the checklist. would be missing:_

- Mic slider (with hover effect)
- systray

_And I find a small issue with the volume slider. I prefer that the volume % would be configurable. For example, if I want it to go up or down by 5% that can be configured. Since the percentage currently seems irrational. The first step is 15% (controlled with pavucontrol)._

**Comment by scorpion-26** [comment]
> Oh. I forgot to tell you yes. Everything works perfect!

Glad to hear that!

> Mic slider (with hover effect)

What do you mean by hover effect? Like reveal on hover or with flyin-support?

> I prefer that the volume % would be configurable.

Now done, configurable via ```AudioScrollSpeed``` in the config. Note however, that there seems to be a precision error, causing it to sometimes jump up by e.g. 4 instead of the configured 5. If that's a problem, let me know.

**Comment by linuxmobile** [comment]
> If that's a problem, let me know.

It's working fine. But even from 0 to 15% is fine. From 15% it goes to 19%, from 19 to 24%, from 24 to 29%, 29% to 33%, from 33 to 38%. But it isn't a real problem. Thx for fix everything!

**Comment by scorpion-26** [comment]
> Mic slider with hover effect to open the slider too.
Now done. Note that you need to set ```AudioInput``` to true and add .mic-icon / .mic-volume to the css. You can see the diff for reference.

**Comment by linuxmobile** [comment]
> Now done. Note that you need to set `AudioInput` to true and add .mic-icon / .mic-volume to the css. You can see the diff for reference.

Everything works fine, but the volume icons now do not appear for me. It's probably because of the font I'm using.

![image](https://user-images.githubusercontent.com/10554636/220932201-c88f5983-9362-46c8-8985-c823f42ea389.png)

**Comment by scorpion-26** [comment]
I use the 'Material Design' Icons from Nerd Font for the volume icons. Since [version 2.3.0](https://github.com/ryanoasis/nerd-fonts/blob/master/changelog.md) the old locations were deprecated in favor of new locations. I use the new ones. So you probably need to update the version of the Nerd Font.

**Comment by scorpion-26** [comment]
> I think you could add a systray. Since I can see the telegram and discord icons.

Finally got around to implementing something. Tested for discord, telegram and steam. With the current implementation there should be icons appearing on the bar out-of-the-box (Even though they are all over the place). Note that the implementation is by no means complete. Bugs/Issues regarding this feature should be posted on the PR I opened, since progress tracking happens there.
